src/Salad.java

```diff
@@ -0,0 +1,73 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+
+public class Salad {
+    private String name;
+    ArrayList<Vegetable> ingredients;
```

Comment:
to avoid mistakes, if ingredients must be initialized, do it in declaration. Object params will be initialized on object constuctor unless param is static - in this case param will be initialized when class is created.